genesis/engine/solvers/base_solver.py

```diff
@@ -36,6 +31,29 @@ def __init__(self, scene: "Scene", sim: "Simulator", options):
     def _add_force_field(self, force_field):
         self._ffs.append(force_field)
 
+    def build(self):
+        self._B = self._sim._B
+
+        g_np = np.asarray(
```

Comment:
If think you should rather completely ignore gravity if `self._init_gravity` is None:

```python
if self._init_gravity is not None:
    gravity = np.asarray(self._init_gravity, dtype=np.float32)
    gravity = np.repeat(g_np[None], self._B, axis=0)

    self._gravity = ti.Vector.field(3, dtype=gs.ti_float, shape=self._B)
    self._gravity.from_numpy(gravity)
```